components/permission-provider/org.wso2.carbon.analytics.permissions/src/main/java/org/wso2/carbon/analytics/permissions/PermissionProvider.java

```diff
@@ -112,5 +112,17 @@
      */
     List<Role> getGrantedRoles(Permission permission) throws PermissionException;
 
+    /**
+     * Get list of roles in a tenant which has a permission granted.
+     *
+     * @param permission
+     * @param username
+     * @return
+     * @throws PermissionException
+     */
+    List<Role> getGrantedRolesOfTenant(Permission permission, String username) throws PermissionException;
```

Comment:
This is defined twice. Any reason to use both Permission and string?

Reply:
At the moment we only need `getGrantedRolesOfTenant(Permission permission, String username)` method. I have added   `getGrantedRolesOfTenant(String permissionID, String username)` in case if needed in the future. If needed I can remove the string one.

Reply:
Let's remove it since it's not needed as of now.

Reply:
Fixed in 6345a1d